### PR TITLE
Running ssh-vm.sh gives instant access to the Harvester VM running for that workspace

### LINKED ISSUE #643: Error: Cannot read property 'sha' of undefined
When attempting to open https://gitpod.io/#https://github.com/nico-abram/rusty-boy-advance/ , I get the following: https://i.imgur.com/uBmFwI4.png

### PR DESCRIPTION
**Pull request by wulfthimm**
## Description

This gives instant access to a running VM on harvester using the command `./dev/preview/ssh-vm.sh`. If the namespace is changed the new SSH connection closes the old forwarding and creates a new connection. This can also be avoided by adding the options "-p PORT -n NAMESPACE". This way a new connection is created using another port keeping the old one.

## Related Issue(s)

Fixes #643

## How to test

Create a new branch and a commit with `/werft with-vm=true` and login to the machine using `./dev/preview/ssh-vm.sh`. Create another branch and a commit like before. After the VM has started you can login with `./dev/preview/ssh-vm.sh -p 8023`. This keeps the connection 8022 and opens a new connection on port 8023 to the new VM.

## Release Notes

```release-note
None
```

### CODE REVIEW

**Comment by roboquat** [comment]
@wulfthimm: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mrsimonemms** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mrsimonemms** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms` in a comment when ready.

Associated issue: *#643*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mads-hartmann** [comment]
@wulfthimm removed Platform as reviews, please re-assign once the conflicts are fixed :)

File: .gitpod.yml
Comment (ArthurSens) [inline .gitpod.yml:35]: What do you think of doing the whole SSH connection in one of the terminal tabs and keeping it running?
Maybe just doing a quick check to verify that the workspace has a related VM, then close this tab if it hasn't

Comment (wulfthimm) [reply .gitpod.yml:35]: That is basically a good idea and would also add some comfort. I am personally a fan of doing it myself. @mads-hartmann What do you thing?

Comment (mads-hartmann) [reply .gitpod.yml:35]: I'm hoping that SSH access will only be needed for debugging issues with the VM, so if everything works smoothly the SSH connection shouldn't be needed most of the time - most likely developers will primarily use kubectl to get the logs etc. for their applications which we're trying to make convenient in https://github.com/gitpod-io/ops/issues/644

For for now, let's keep it as it, where developers have to run a script whenever they need SSH access ☺️

**Review by ArthurSens — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 8a237659ca80d76ef10a0fc8279d3d016ca72249</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7756#pullrequestreview-860799805" title="Approved">ArthurSens</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mrsimonemms** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms` in a comment when ready.

Associated issue: *#643*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: .gitpod.yml
Comment (ArthurSens) [reply .gitpod.yml:35]: I believe that just creating a terminal for the SSH connection already open wouldn't hurt as well 😅 

No problems doing that later though

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

File: dev/preview/install-vm-ssh-keys.sh
Comment (mads-hartmann) [inline dev/preview/install-vm-ssh-keys.sh:18]: Use `log` instead for consistent formatting :)

File: dev/preview/ssh-proxy-command.sh
Comment (mads-hartmann) [inline dev/preview/ssh-proxy-command.sh:22]: Was hoping to use `nc` here to check for the "readiness" of port $PORT in a while loop and sleep 1 for each failure. With 5 seconds it might be fragile.

**Review by mads-hartmann — COMMENTED** [review]
Left two minor suggestions

**Comment by wulfthimm** [comment]
SSH access does not work at the moment. I created an [issue for that](https://github.com/gitpod-io/gitpod/issues/7799).

File: dev/preview/install-vm-ssh-keys.sh
Comment (wulfthimm) [reply dev/preview/install-vm-ssh-keys.sh]: Done

File: dev/preview/ssh-proxy-command.sh
Comment (wulfthimm) [reply dev/preview/ssh-proxy-command.sh]: Done

**Comment by wulfthimm** [comment]
> You can pick my commit [6926c57](https://github.com/gitpod-io/gitpod/commit/6926c575f4485b4275cb151eeba0320c23347905)
> 
> it will change vm-ssh port to 2200

Many thanks, @iQQBot !

**Review by mads-hartmann — APPROVED** [review]
The code changes look good to me! Leaving the hold label so you can merge when you're comfortable.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 125c7912277c85b5a19764d87e12b03b3b4256ef</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7756#pullrequestreview-860799805" title="Approved">ArthurSens</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7756#pullrequestreview-863317816" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mrsimonemms** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms` in a comment when ready.

Associated issue: *#643*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ArthurSens** [comment]
If there is nothing left to review here, could we merge https://github.com/gitpod-io/gitpod/pull/7835 and try to get this one merged as well? :)

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Review by mads-hartmann — APPROVED** [review]
🚀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 54ae6d3768eeb86d10982a2f478f937145eedf9b</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7756#pullrequestreview-860799805" title="Approved">ArthurSens</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7756#pullrequestreview-863401028" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mrsimonemms** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms` in a comment when ready.

Associated issue: *#643*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7756#pullrequestreview-860799805" title="Approved">ArthurSens</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7756#pullrequestreview-863401028" title="Approved">mads-hartmann</a>*

Associated issue: *#643*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
